doc/nvim-treesitter.txt

```diff
@@ -449,228 +449,236 @@ Note: This is highly experimental, and folding can break on some types of
+Constructor calls and definitions: `{}` in Lua, and Java constructors.
 
 								  *hl-TSError*
 `TSError`
-For syntax/parser errors.
+Syntax/parser errors. This might highlight large sections of code while the
+user is typing still incomplete code, use a sensible highlight.
 
 							      *hl-TSException*
 `TSException`
-For exception related keywords.
+Exception related keywords: `try`, `except`, `finally` in Python.
 
 								  *hl-TSField*
 `TSField`
-For fields.
+Object and struct fields.
 
 								  *hl-TSFloat*
 `TSFloat`
-For floats.
+Floating-point number literals.
 
 							       *hl-TSFunction*
 `TSFunction`
-For function (calls and definitions).
+Function calls and definitions.
 
 							    *hl-TSFuncBuiltin*
 `TSFuncBuiltin`
-For builtin functions: `table.insert` in Lua.
+Built-in functions: `print` in Lua.
 
 							      *hl-TSFuncMacro*
 `TSFuncMacro`
-For macro defined fuctions (calls and definitions): each `macro_rules` in
+Macro defined functions (calls and definitions): each `macro_rules` in
 Rust.
 
 								*hl-TSInclude*
 `TSInclude`
-For includes: `#include` in C, `use` or `extern crate` in Rust, or `require`
-in Lua.
+File or module inclusion keywords: `#include` in C, `use` or `extern crate` in
+Rust.
 
 								*hl-TSKeyword*
 `TSKeyword`
-For keywords that don't fall in previous categories.
+Keywords that don't fit into other categories.
 
 							*hl-TSKeywordFunction*
 `TSKeywordFunction`
-For keywords used to define a fuction.
+Keywords used to define a function: `function` in Lua, `def` and `lambda` in
+Python.
 
 							*hl-TSKeywordOperator*
 `TSKeywordOperator`
-for operators that are English words, e.g. `and`, `as`, `or`.
+Unary and binary operators that are English words: `and`, `or` in Python;
+`sizeof` in C.
 
 							  *hl-TSKeywordReturn*
 `TSKeywordReturn`
-for the `return` and `yield` keywords.
+Keywords like `return` and `yield`.
 
 								  *hl-TSLabel*
 `TSLabel`
-For labels: `label:` in C and `:label:` in Lua.
+GOTO labels: `label:` in C, and `::label::` in Lua.
 
 								 *hl-TSMethod*
 `TSMethod`
-For method calls and definitions.
+Method calls and definitions.
 
 							      *hl-TSNamespace*
 `TSNamespace`
-For identifiers referring to modules and namespaces.
+Identifiers referring to modules and namespaces.
 
 								     *hl-None*
 `TSNone`
-For no highlighting.
+No highlighting.
 
 								 *hl-TSNumber*
 `TSNumber`
-For all numbers
+Any numeric literal that doesn't fit into other categories.
 
 							       *hl-TSOperator*
 `TSOperator`
-For any operator: `+`, but also `->` and `*` in C.
+Binary or unary operators: `+`, and also `->` and `*` in C.
 
 							      *hl-TSParameter*
 `TSParameter`
-For parameters of a function.
+Parameters of a function.
 
 						     *hl-TSParameterReference*
 `TSParameterReference`
-For references to parameters of a function.
+References to parameters of a function.
 
 							       *hl-TSProperty*
 `TSProperty`
 Same as `TSField`.
```

Comment:
we should find an example for this, I'm not myself sure about the difference :smile:

Reply:
I _think_ this highlight group isn't being used, since it requires more logic than other highlights (besides checking if it's an identifier it'd have to check whether it's listed in the function signature). 

Even if the queries checked that, it does seem inconsistent with other highlight groups considering that definitions and usage are highlighted the same for functions and types. 🤔